crates/metaslang/cst/src/query/parser.rs

```diff
@@ -41,13 +44,38 @@ pub(super) fn parse_query<T: KindTypes>(input: &str) -> Result<ASTNode<T>, Query
         .parse(input)
         .finish()
         .map(|(_, query)| query)
-        .map_err(|e| QueryError {
-            message: e.to_string(),
-            row: 0,
-            column: 0,
+        .map_err(|e| {
+            let text_index = compute_row_and_column(e.errors[0].0, input);
+            QueryError {
+                message: e.to_string(),
+                row: text_index.line,
+                column: text_index.column,
+            }
         })
 }
 
+fn compute_row_and_column(target: &str, from: &str) -> TextIndex {
+    let target_offset = from.offset(target);
```

Comment:
`offset` is implemented by `nom` and works by [subtracting the pointers](https://github.com/rust-bakery/nom/blob/main/src/traits.rs#L603-L610) to the string references. So, different substrings, even if they have the same content will not produce the same byte offset.

Reply:
Btw, it will panic if you pass the parameters reversed, because of overflow when subtracting the pointers casted as `usize`. But this should never happen as `target` will always be a substring of `from`/`input`.

Reply:
I missed that. Thanks for pointing it out.